src/main/scala/com/microsoft/ml/spark/lightgbm/params/FObjTrait.scala

```diff
@@ -0,0 +1,17 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.ml.spark.lightgbm.params
+
+import com.microsoft.ml.spark.lightgbm.dataset.LightGBMDataset
+
+trait FObjTrait extends Serializable {
+  /**
+    * User defined objective function, returns gradient and second order gradient
+    *
+    * @param predictions untransformed margin predicts
+    * @param trainingData training data
+    * @return List with two float array, correspond to grad and hess
+    */
+  def getGradient(predictions: Array[Array[Double]], trainingData: LightGBMDataset): List[Array[Float]]
```

Comment:
Might want to make this return two objects instead of a list. Also in general the grad and hessian wont be the same shape...

Reply:
interesting idea, in xgboost I see they return one list:
https://github.com/dmlc/xgboost/blob/4224c08cacceba3f83f90e387c07aa6205a83bfa/jvm-packages/xgboost4j/src/main/scala/ml/dmlc/xgboost4j/scala/ObjectiveTrait.scala#L32
I can return two objects but for those familiar with xgboost it might be harder to get used to a slightly different API

Reply:
done!  changes it to return a tuple